data_management_scripts/cleanup_dataset_matrixtables.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3  # noqa: EXE001
+
+import logging
+import subprocess
+import sys
+from datetime import datetime
+
+import click
+from cpg_utils.config import get_config
+from google.cloud import storage
+
+GENOME_PREFIX = 'mt/'
+EXOME_PREFIX = 'exome/mt/'
+
+
+def get_dataset_mt_timestamps(
+    dataset: str,
+    bucket: storage.Bucket,
+    prefix: str,
+):
+    """Finds the .mt directories and their create times in the seqr-main bucket"""
+    mt_timestamps: dict[str, datetime] | None = {}
+    try:
+        mt_folders = (
+            subprocess.run(  # Use subprocess because list_blobs lists everything inside each .mt folder
+                args=['gsutil', 'ls', f'gs://{bucket.name}/{prefix}'],
+                check=True,
+                capture_output=True,
+                text=True,
+            )
+            .stdout.strip()
+            .split('\n')
+        )
+    except subprocess.CalledProcessError:
+        logging.info(f'{dataset}:: No .mt folders found in {prefix}')
+        return mt_timestamps
+
+    mt_folders = [
+        mt_folder.removeprefix(f'gs://{bucket.name}/') for mt_folder in mt_folders
+    ]
+
+    for mt_folder in mt_folders:
+        logging.info(f'{dataset}:: Found: {mt_folder}')
+        blob = bucket.get_blob(mt_folder)
+        if not blob:  # If the .mt folder is not a blob, use the _SUCCESS file
+            blob = bucket.get_blob(f'{mt_folder}_SUCCESS')
+        mt_timestamps[mt_folder] = blob.time_created
+
+    return mt_timestamps
+
+
+def get_mt_folders_to_delete(
+    dataset: str,
+    mt_timestamps_dict: dict[str, datetime],
+):
+    """Takes the list of .mt directories and keeps only the most recently created one"""
+    dataset_mts = sorted(mt_timestamps_dict.items(), key=lambda x: x[1])
```

Comment:
Sorting by timestamp is fine, but we should also be happy ripping this from the names. The MT naming convention is `BUCKET/mt/RANDOM-HASH_NUMBER-OF-SAMPLES-PROJECT.mt` - where presumably the latest will always have the highest number of included samples